charms/knative-eventing/src/charm.py

```diff
@@ -4,101 +4,85 @@
 #
 # Learn more at: https://juju.is/docs/sdk
 
-"""Charm the service.
-
-Refer to the following post for a quick-start guide that will help you
-develop a new k8s charm using the Operator Framework:
-
-    https://discourse.charmhub.io/t/4208
-"""
+"""A Juju charm for Knative Eventing."""
 
+import glob
 import logging
-
+import traceback
+from pathlib import Path
+
+from k8s_resource_handler.exceptions import ErrorWithStatus
```

Comment:
I forgot to bring this up during the other resource handler/helpers discussions - I pretty much just renamed `CheckFailed` unilaterally to something more generic because the idea of an exception with a status got reused elsewhere.  Does this feel like the right name/functionality?  Should we add a CheckFailed in as well?  Feel free to question/trash :)

Reply:
I like `ErrorWithStatus` also because it gives the user to option to ignore it if it does not apply.